t/t3402-rebase-merge.sh

```diff
@@ -46,6 +46,33 @@ test_expect_success setup '
 	git checkout -b test-merge side
 '
 
+test_expect_success 'setup for --ignore-whitespace' '
```

Comment:
@tgummerer @newren I feel that this looks 'ugly' and prefer moving them to a separate file. I know both backends do not fully support the flag but we can add a NEEDSWORK or test_must_fail to indicate improving the test/command in future. 't3431-rebase-compatibility-options.sh` could be the name indeed.

Reply:
Yeah looking at this it's quite ugly.  I was hoping we could integrate it better with the existing test code.

Especially the test below for the am backend doesn't make all that much sense in this test file (though I thought we are just dropping that?).

If you feel strongly about having this in a new test file I don't mind too much, and we can see how people on the mailing list feel about that.

Sorry about the back and forth here.